src/Microsoft.NetCore.Analyzers/UnitTests/Tasks/UseValueTaskCorrectlyTests.cs

```diff
@@ -0,0 +1,1274 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Tasks.UnitTests
+{
+    public class UseValueTasksCorrectlyTests : DiagnosticAnalyzerTestBase
+    {
+        protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer() => new UseValueTasksCorrectlyAnalyzer();
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer() => new UseValueTasksCorrectlyAnalyzer();
+
+        private static readonly ParseOptions s_csharpPreviewParseOptions = new CSharpParseOptions(CodeAnalysis.CSharp.LanguageVersion.Preview);
+        private static readonly ParseOptions s_basicPreviewParseOptions = new VisualBasicParseOptions(CodeAnalysis.VisualBasic.LanguageVersion.Latest);
+
+        private static string CSharpBoilerplate(string s) => s +
+            @"
+            internal static class Helpers
```

Comment:
Nit: Can consider moving these to https://github.com/dotnet/roslyn-analyzers/tree/master/src/Test.Utilities/MinimalImplementations so can be reused by other tests. It seems these will potentially be used by multiple analyzers in future.